File: 2792-neighboring-bitwise-xor/2792-neighboring-bitwise-xor.c
bool doesValidArrayExist (int* derived, int derivedSize) {  
    int org [derivedSize];

    for (int i = 1; i < derivedSize; i++) {
        org [i] = derived [i - 1] ^ org [i - 1];
    }

    for (int i = 0; i < derivedSize; i++) {
        int index = (i + 1) % derivedSize;
        if (derived [i] != (org [i] ^ org [index])) return false;
    }
    return true;
}

File: 2792-neighboring-bitwise-xor/2792-neighboring-bitwise-xor.java
class Solution {
    public boolean doesValidArrayExist (int [] derived) {
        int n = derived.length;
        int [] org = new int [n];

        for (int i = 1; i < n; i++) {
            org [i] = derived [i - 1] ^ org [i - 1];
        }

        for (int i = 0; i < n; i++) {
            int index = (i + 1) % n;
            if (derived [i] != (org [i] ^ org [index])) return false;
        }

        return true;
    }
}